### PR TITLE
Configure test coverage

### PR DESCRIPTION
**Pull request by dblandin**
Generate test coverage data during test suite run and report to Code Climate.

@codeclimate/review :mag_right:

### CODE REVIEW

**Comment by wfleming** [comment]
👍

File: package.json [hunk 1]
```diff
@@ -4,14 +4,15 @@
   "description": "Code Climate test reporter client for javascript projects",
   "main": "index.js",
   "scripts": {
-    "test": "mocha"
+    "test": "istanbul cover --report lcov _mocha"
```
Comment (dblandin) [inline package.json]: The need for `_mocha` here is odd but it is a known issue: https://github.com/gotwarlost/istanbul/issues/44

File: .travis.yml [hunk 2]
```diff
@@ -11,3 +11,6 @@ node_js:
 cache:
   directories:
     - node_modules
+after_script:
+  - npm install -g codeclimate-test-reporter
```
Comment (dblandin) [inline .travis.yml]: I wouldn't expect this to be necessary after configuring the `code_climate` addon above. It doesn't look like the package is installed automatically, as I would expect.

https://travis-ci.org/codeclimate/javascript-test-reporter/jobs/135634591#L253

```
$ codeclimate-test-reporter < ./coverage/lcov.info
/home/travis/build.sh: line 45: codeclimate: command not found
```

Comment (bkyryliuk) [inline .travis.yml]: you can ddd codeclimate-test-reporter to the devDependencies.
Example here:
https://github.com/airbnb/caravel/blob/master/caravel/assets/package.json#L95

And this is a sample PR where I've set up it to work:
https://github.com/airbnb/caravel/pull/1046

Only issue I couldn't fix was using the travis env variables. codeclimate-test-reporter doesn't see them

**Pull request merged**